hedera-node/src/main/java/com/hedera/services/state/merkle/internals/AbstractMerkleMapValueListNode.java

```diff
@@ -0,0 +1,144 @@
+package com.hedera.services.state.merkle.internals;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+
+import com.swirlds.common.FastCopyable;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import com.swirlds.common.merkle.utility.AbstractMerkleLeaf;
+import com.swirlds.common.merkle.utility.Keyed;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Provides implementation assistance for a {@link MapValueListNode} that lives in a
+ * {@link com.swirlds.merkle.map.MerkleMap} whose keys are both {@link SelfSerializable} and {@link FastCopyable}.
+ */
+public abstract class AbstractMerkleMapValueListNode<
+		K extends SelfSerializable & FastCopyable,
+		S extends AbstractMerkleMapValueListNode<K, S>>
+		extends AbstractMerkleLeaf implements MerkleLeaf, Keyed<K>, MapValueListNode<K, S> {
+
+	private static final int PREV_INDEX = 0;
+	private static final int CUR_INDEX = 1;
+	private static final int NEXT_INDEX = 2;
+
+	private K[] keys;
```

Comment:
Why is there an array?

Reply:
Can we have 3 key type variables instead of array ?